File: app/src/main/java/com/example/soroushprofile/prefrences/widgets/SoroushPreferenceCategory.java
package com.example.soroushprofile.prefrences.widgets;

import android.content.Context;
import android.text.TextUtils;
import android.util.AttributeSet;
import android.view.View;
import android.widget.TextView;

import androidx.preference.PreferenceCategory;
import androidx.preference.PreferenceViewHolder;

import com.example.soroushprofile.R;

public class SoroushPreferenceCategory extends PreferenceCategory {

    private TextView rightSummary;
    private CharSequence summary;
    private View.OnClickListener mSummaryClickListener;

    public SoroushPreferenceCategory(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
        initialize();
    }

    public SoroushPreferenceCategory(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        initialize();
    }

    public SoroushPreferenceCategory(Context context, AttributeSet attrs) {
        super(context, attrs);
        initialize();
    }

    public SoroushPreferenceCategory(Context context) {
        super(context);
        initialize();
    }

    public void setOnSummaryClickListener(View.OnClickListener summaryClickListener) {
        mSummaryClickListener = summaryClickListener;
    }

    private void initialize() {
        setLayoutResource(R.layout.preference_right_summary_widget);
    }

    @Override
    public void onBindViewHolder(PreferenceViewHolder holder) {
        super.onBindViewHolder(holder);
        rightSummary = (TextView) holder.findViewById(R.id.right_summary);
        setSummary(summary);
    }

    @Override
    public void setSummary(CharSequence summary) {
        super.setSummary(null);
        this.summary = summary;

        if (this.rightSummary != null) {
            this.rightSummary.setText(summary);
            this.rightSummary.setOnClickListener(mSummaryClickListener);
            this.rightSummary.setVisibility(TextUtils.isEmpty(summary) ? View.GONE : View.VISIBLE);
        }
    }
}

File: app/src/main/java/com/example/soroushprofile/prefrences/widgets/SoroushPreferenceCategory.kt
package com.example.soroushprofile.prefrences.widgets

import android.content.Context
import android.text.TextUtils
import android.util.AttributeSet
import android.view.View
import android.widget.TextView

import androidx.preference.PreferenceCategory
import androidx.preference.PreferenceViewHolder

import com.example.soroushprofile.R

class SoroushPreferenceCategory : PreferenceCategory {

    private var rightSummary: TextView? = null
    private var mSummary: CharSequence? = null
    private var mSummaryClickListener: View.OnClickListener? = null

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int, defStyleRes: Int) : super(context, attrs, defStyleAttr, defStyleRes) {
        initialize()
    }

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        initialize()
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        initialize()
    }

    constructor(context: Context) : super(context) {
        initialize()
    }

    fun setOnSummaryClickListener(summaryClickListener: View.OnClickListener) {
        mSummaryClickListener = summaryClickListener
    }

    private fun initialize() {
        layoutResource = R.layout.preference_right_summary_widget
    }

    override fun onBindViewHolder(holder: PreferenceViewHolder) {
        super.onBindViewHolder(holder)
        rightSummary = holder.findViewById(R.id.right_summary) as TextView
        setSummary(mSummary)
    }

    override fun setSummary(summary: CharSequence?) {
        super.setSummary(null)
        this.mSummary = summary

        if (this.rightSummary != null) {
            this.rightSummary!!.text = summary
            this.rightSummary!!.setOnClickListener(mSummaryClickListener)
            this.rightSummary!!.visibility = if (TextUtils.isEmpty(summary)) View.GONE else View.VISIBLE
        }
    }
}
